sql/catalyst/src/main/java/org/apache/spark/sql/BaseRow.java

```diff
@@ -155,27 +155,6 @@ public int fieldIndex(String name) {
     throw new UnsupportedOperationException();
   }
 
-  /**
-   * A generic version of Row.equals(Row), which is used for tests.
-   */
-  @Override
```

Comment:
Because `Row` is a trait, UnsafeRow and SpecificRow are both in Java, they can not inherit some default implementations from `Row`, so created BaseRow in Java for them. Right now, we have `InternalRow`, will be clean these in another PR.